### PR TITLE
[QNN EP] Qnn batchnorm Op support

### PR DESCRIPTION
**Pull request by HectorSVC**
### Description
Support BatchNorm Op in Qnn EP
Node Unit group support for BatchNorm, Exp ops

### Motivation and Context
Enable more models.

### CODE REVIEW

**Comment by HectorSVC** [comment]
Will add QDQ test later.

File: onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc [hunk 1]
```diff
@@ -352,6 +352,30 @@ bool InstanceNormalizationNodeGroupSelector::Check(const GraphViewer& graph_view
...
+  if (!CheckQDQNodes(graph_viewer, node, dq_nodes, q_nodes)) {
+    return false;
+  }
+
+  int32_t dt_input = dq_nodes[0]->InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+  int32_t dt_scale = dq_nodes[1]->InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+  int32_t dt_output = q_nodes[0]->OutputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+  if (dt_input != dt_output) {
+    return false;
+  }
```
Comment (adrianlizarraga) [inline onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc:368]: Should there also be a check for the bias input's type, or is it unnecessary?

File: onnxruntime/core/providers/qnn/builder/opbuilder/batch_norm_op_builder.cc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    ORT_RETURN_IF_NOT(scale_shape.size() != 1 || scale_shape[0] != num_channels,
+                      "QNN BatchNorm input 1 (scale) must have 1D shape [channel].");
+
+    std::vector<uint32_t> bias_shape;
+    ORT_RETURN_IF_NOT(qnn_model_wrapper.GetOnnxShape(inputs[2].node_arg, bias_shape), "Cannot get shape of input 2 (bias)");
+    ORT_RETURN_IF_NOT(bias_shape.size() != 1 || bias_shape[0] != num_channels,
+                      "QNN BatchNorm input 2 (bias) must have 1D shape [channel].");
+
+    std::vector<uint32_t> mean_shape;
+    ORT_RETURN_IF_NOT(qnn_model_wrapper.GetOnnxShape(inputs[3].node_arg, mean_shape), "Cannot get shape of input 3 (mean)");
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/batch_norm_op_builder.cc:69]: I wonder if there's any reason to validate the shapes of inputs 3 and 4 if they're not used by QNN. Maybe we omit these checks on input_mean and input_var?

File: onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc (hunk 1 above)
Comment (HectorSVC) [reply onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc:368]: Right now, it's int8 or uint8 from Ort quantization script, actually, it supports int32 for better accuracy like what we did for Conv. Don't want to make it too tight.

---
In reply to: [1149659300](https://github.com/microsoft/onnxruntime/pull/15222#discussion_r1149659300) [](ancestors = 1149659300)

File: onnxruntime/core/providers/qnn/builder/opbuilder/batch_norm_op_builder.cc (hunk 2 above)
Comment (HectorSVC) [reply onnxruntime/core/providers/qnn/builder/opbuilder/batch_norm_op_builder.cc:69]: From Onnx spec, the mean & var inputs are required instead of optional, so we have to do some validation to cover it. Also there are some UTs for these case. Need to get the UTs pass.

---
In reply to: [1149666966](https://github.com/microsoft/onnxruntime/pull/15222#discussion_r1149666966) [](ancestors = 1149666966)

Comment (adrianlizarraga) [reply onnxruntime/core/providers/qnn/builder/opbuilder/batch_norm_op_builder.cc:69]: Sounds good.

**Review by adrianlizarraga — APPROVED** [review]
:shipit:
